src/Shared/Model/Repository.cs

```diff
@@ -0,0 +1,166 @@
+
+        [JsonProperty(PropertyName = "licenses", NullValueHandling = NullValueHandling.Ignore)]
+        public List<License>? Licenses { get; set; }
+
+        [JsonProperty(PropertyName = "linked_data", NullValueHandling = NullValueHandling.Ignore)]
+        public LinkedData? LinkedData { get; set; }
+
+        [JsonProperty(PropertyName = "maintainers", NullValueHandling = NullValueHandling.Ignore)]
+        public List<User>? Maintainers { get; set; }
+
+        [JsonProperty(PropertyName = "openissues_count", NullValueHandling = NullValueHandling.Ignore)]
+        public int? OpenIssuesCount { get; set; }
+
+        [JsonProperty(PropertyName = "owner", NullValueHandling = NullValueHandling.Ignore)]
+        public User? Owner { get; set; }
+
+        [JsonProperty(PropertyName = "parent", NullValueHandling = NullValueHandling.Ignore)]
+        public string? Parent { get; set; }
+
+        [JsonProperty(PropertyName = "purl", NullValueHandling = NullValueHandling.Ignore)]
+        public string? Purl { get; set; }
+
+        [JsonProperty(PropertyName = "pushed_at", NullValueHandling = NullValueHandling.Ignore)]
+        public DateTimeOffset? PushedAt { get; set; }
+
+        [JsonProperty(PropertyName = "rank", NullValueHandling = NullValueHandling.Ignore)]
+        public double? Rank { get; set; }
+
+        [JsonProperty(PropertyName = "size", NullValueHandling = NullValueHandling.Ignore)]
+        public long? Size { get; set; }
+
+        [JsonProperty(PropertyName = "stakeholders_count", NullValueHandling = NullValueHandling.Ignore)]
+        public int? StakeholdersCount { get; set; }
+
+        [JsonProperty(PropertyName = "type", NullValueHandling = NullValueHandling.Ignore)]
+        public string? Type { get; set; }
+
+        [JsonProperty(PropertyName = "updated_at", NullValueHandling = NullValueHandling.Ignore)]
+        public DateTimeOffset? UpdatedAt { get; set; }
+
+        [JsonProperty(PropertyName = "url", NullValueHandling = NullValueHandling.Ignore)]
+        public string? Uri { get; set; }
+
+        public async Task<CSTRepository?> ExtractRepositoryMetadata(PackageURL purl)
+        {
+            if (purl is not null && purl.ToString() is string purlString && !string.IsNullOrWhiteSpace(purlString))
+            {
+                if (purl.Type != "github")
+                {
+                    Logger.Warn("Only github repos are handled currently");
+                    return this;
+                }
+
+                return await FetchGithubRepositoryMetadata(purl);
+            }
+            return null;
+        }
+
+        private async Task<CSTRepository> FetchGithubRepositoryMetadata(PackageURL purl)
+        {
+            try
+            {
+                var github = new GitHubClient(new ProductHeaderValue("OSSGadget"));
+                var ghRepository = await github.Repository.Get(purl.Namespace, purl.Name);
+
+                if (ghRepository is null) { return null; }
```

Comment:
If you're returning null this actually returns `CSTRepository?` not `CSTRepository`